pkg/security/probe/probe_kernel_file_windows.go

```diff
@@ -135,105 +101,119 @@ The Parameters.Create.FileAttributes and Parameters.Create.EaLength members are
+		wp.discardedFileHandles.Add(fileObjectPointer(fileObject), struct{}{})
 		wp.stats.fileCreateSkippedDiscardedPaths++
-		return nil, errDiscardedPath
+		return 0, fileCache{}, errDiscardedPath
 	}
 
-	ca.userFileName = wp.mustConvertDrivePath(ca.fileName)
-	if _, ok := wp.discardedUserPaths.Get(ca.userFileName); ok {
+	userFileName := wp.mustConvertDrivePath(fileName)
+	if _, ok := wp.discardedUserPaths.Get(userFileName); ok {
 		wp.stats.fileCreateSkippedDiscardedPaths++
-		return nil, errDiscardedPath
+		return 0, fileCache{}, errDiscardedPath
 	}
 
 	if _, ok := wp.discardedBasenames.Get(basename); ok {
-		wp.discardedFileHandles.Add(fileObjectPointer(ca.fileObject), struct{}{})
+		wp.discardedFileHandles.Add(fileObjectPointer(fileObject), struct{}{})
 		wp.stats.fileCreateSkippedDiscardedBasenames++
-		return nil, errDiscardedPath
+		return 0, fileCache{}, errDiscardedPath
 	}
 
 	// lru is thread safe, has its own locking
 	fc := fileCache{
-		fileName:     ca.fileName,
-		userFileName: ca.userFileName,
+		fileName:     fileName,
+		userFileName: userFileName,
+		baseName:     basename,
 	}
-	if wp.filePathResolver.Add(ca.fileObject, fc) {
+	if wp.filePathResolver.Add(fileObject, fc) {
 		wp.stats.fileNameCacheEvictions++
 	}
 	// if we get here, we have a new file handle.  Remove it from the discarder cache in case
 	// we missed the close notification
-	wp.discardedFileHandles.Remove(fileObjectPointer(ca.fileObject))
+	wp.discardedFileHandles.Remove(fileObjectPointer(fileObject))
 
-	return ca, nil
+	return fileObject, fc, nil
 }
 
-func (wp *WindowsProbe) parseCreateNewFileArgs(e *etw.DDEventRecord) (*createNewFileArgs, error) {
-	ca, err := wp.parseCreateHandleArgs(e)
-	if err != nil {
-		return nil, err
-	}
-	return (*createNewFileArgs)(ca), nil
+func (wp *WindowsProbe) onCreateHandleArgs(e *etw.DDEventRecord) {
+	// for the createHandle, we don't need to do anything other than ensure that the
+	// file is processed and added to the cache.
+	_, _, _ = wp.parseCreateHandleArgs(e)
 }
 
-// nolint: unused
-func (ca *createHandleArgs) string(t string) string {
-	var output strings.Builder
-
-	output.WriteString(t + " PID: " + strconv.Itoa(int(ca.ProcessID)) + "\n")
-	output.WriteString("         Name: " + ca.fileName + "\n")
-	output.WriteString("         Opts: " + strconv.FormatUint(uint64(ca.createOptions), 16) + " Share: " + strconv.FormatUint(uint64(ca.shareAccess), 16) + "\n")
-	output.WriteString("         OBJ:  " + strconv.FormatUint(uint64(ca.fileObject), 16) + "\n")
-
-	return output.String()
-}
-
-// nolint: unused
-func (ca *createHandleArgs) String() string {
-	return ca.string("CREATE")
-}
-
-// nolint: unused
-func (ca *createNewFileArgs) String() string {
-	return (*createHandleArgs)(ca).string("CREATE_NEW_FILE")
+func (wp *WindowsProbe) onCreateNewFileArgs(e *etw.DDEventRecord) *model.Event {
+	fo, fc, err := wp.parseCreateHandleArgs(e)
+	if err != nil {
+		return nil
+	}
+	ev, err := wp.eventCache.Get()
```

Comment:
It seems it is more or less https://pkg.go.dev/sync#Pool ? should we use a pool instead ?